aten/src/ATen/PythonTorchFunctionTLS.h

```diff
@@ -20,11 +22,15 @@ struct TORCH_API PythonTorchFunctionTLS {
 
  private:
   // The mode TLS is split into
-  //   - disabled_, which says whether or not to disable all torch function
-  //   modes
+  //   - disable_subclass_, which says whether or not to disable torch function
+  //     on subclasses
+  //   - disable_all_, which says whether or not to disable torch function on
+  //     both subclasses and modes. If this is true, it takes precedence over
+  //     disable_subclass
```

Comment:
Turn this into an enum, OR rep disable subclass and mode separately (in both cases, make illegal states unrepresentable)